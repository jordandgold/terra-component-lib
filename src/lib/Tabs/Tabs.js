import React, { Component } from "react";
import "./Tabs.scss";
import PropTypes from "prop-types";

class Tabs extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selected: this.props.selected
    };
  }

  handleTabChange = index => {
    this.setState({
      selected: index
    });
  };

  render() {
    const variantClass = this.props.fullWidth ? "ter-tabs--full-width" : "";

    return (
      <nav
        className={`ter-tabs ${variantClass}`}
        role="navigation"
        data-controls="#tabs-one"
      >
        <ul className="ter-tabs__list" role="tablist">
          {this.props.children.map((elem, index) => {
            let selectedClass =
              index === this.state.selected ? "is-selected" : "";
            return (
              <li
                className={`ter-tabs__list-item ${selectedClass}`}
                key={index}
              >
                <button onClick={() => this.handleTabChange(index)}>
                  {elem.props.name}
                </button>
              </li>
            );
          })}
        </ul>
        <div className="ter-tabs__content">
          {this.props.children[this.state.selected]}
        </div>
      </nav>
    );
  }
}

export const TabsPanel = props => {
  return <div>{props.children}</div>;
};

export default Tabs;

Tabs.propTypes = {
  selected: PropTypes.number.isRequired,
  fullWidth: PropTypes.bool
};
