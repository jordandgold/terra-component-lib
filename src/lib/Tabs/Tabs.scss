@import "../global-styles/terra.scss";

.ter-tabs {
  @include for-size(tablet-landscape-down) {
    display: block;
    overflow-x: auto;
    overflow-y: hidden;
    white-space: nowrap;
  }
  &__list {
    list-style: none;
    margin: 0;
    padding: 0;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    margin-bottom: spacing("sm");
    border-bottom: $nav-tabs-border-size solid $nav-tabs-border-color;
    @include for-size(tablet-landscape-down) {
      display: inline-block;
    }
    > li {
      padding: 0;
      margin-right: 0;
    }
  }
  &__list-item {
    padding: 0;
    margin: 0 0 -2px 0;
    button {
      @include button-base();
      color: $nav-tabs-link-color;
      display: block;
      background-color: $nav-tabs-link-bg;
      font-weight: $nav-tabs-font-weight;
      border-left: $nav-tabs-border-size solid
        transparentize($nav-tabs-border-color, 0.75);
      border-top: $nav-tabs-border-size solid
        transparentize($nav-tabs-border-color, 0.75);
      border-right: 0px;
      border-bottom: $nav-tabs-border-size solid $nav-tabs-link-bg;
      &:hover {
        cursor: pointer;
        color: $nav-tabs-link-hover;
      }
    }
    &:last-child {
      button {
        border-right: $nav-tabs-border-size solid
          transparentize($nav-tabs-border-color, 0.75);
      }
    }
    &.is-selected {
      button {
        border-left: $nav-tabs-border-size solid $nav-tabs-border-color;
        border-top: $nav-tabs-border-size solid $nav-tabs-border-color;
        border-right: $nav-tabs-border-size solid $nav-tabs-border-color;
        border-bottom: 0px;
        color: $nav-tabs-link-hover;
        background-color: $nav-tabs-active-link-bg;
      }
    }
    &.is-selected + li {
      > button {
        border-left: 0px;
      }
    }
  }
  &--full-width {
    .ter-tabs {
      &__list {
        justify-content: space-between;
      }
      &__list-item {
        flex-grow: 1;
        button {
          width: 100%;
        }
      }
    }
  }
}
