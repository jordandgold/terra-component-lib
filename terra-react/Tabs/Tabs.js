import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import React, { Component } from "react";
import "./Tabs.scss";

var Tabs =
/*#__PURE__*/
function (_Component) {
  _inherits(Tabs, _Component);

  function Tabs(props) {
    var _this;

    _classCallCheck(this, Tabs);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Tabs).call(this, props));

    _this.handleTabChange = function (index) {
      _this.setState({
        selected: index
      });
    };

    _this.state = {
      selected: _this.props.selected || 0
    };
    return _this;
  }

  _createClass(Tabs, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var variantClass = this.props.fullWidth ? "ter-tabs--full-width" : "";
      return React.createElement("nav", {
        className: "ter-tabs ".concat(variantClass),
        role: "navigation",
        "data-controls": "#tabs-one"
      }, React.createElement("ul", {
        className: "ter-tabs__list",
        role: "tablist"
      }, this.props.children.map(function (elem, index) {
        var selectedClass = index === _this2.state.selected ? "is-selected" : "";
        return React.createElement("li", {
          className: "ter-tabs__list-item ".concat(selectedClass),
          key: index
        }, React.createElement("button", {
          onClick: function onClick() {
            return _this2.handleTabChange(index);
          }
        }, elem.props.name));
      })), React.createElement("div", {
        className: "ter-tabs__content"
      }, this.props.children[this.state.selected]));
    }
  }]);

  return Tabs;
}(Component);

export var TabsPanel = function TabsPanel(props) {
  return React.createElement("div", null, props.children);
};
export default Tabs;