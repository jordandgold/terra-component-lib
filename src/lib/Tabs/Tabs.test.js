import React from "react";
import { shallow } from "enzyme";
import Tabs, { TabsPanel } from "./Tabs";

describe("Tabs", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Tabs selected={1} fullWidth={false}>
        <TabsPanel name="First">
          <h3>First Tab</h3>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
            lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies
            mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit
            sollicitudin.
          </p>
        </TabsPanel>
        <TabsPanel name="Second">
          <h3>Second Tab</h3>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
            lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies
            mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit
            sollicitudin.
          </p>
        </TabsPanel>
        <TabsPanel name="Third">
          <h3>Third Tab</h3>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
            lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies
            mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit
            sollicitudin.
          </p>
        </TabsPanel>
      </Tabs>
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleTabChange", () => {
    const mockIndex = 2;

    it("Should set state", () => {
      wrapper.instance().handleTabChange(mockIndex);
      expect(wrapper.state().selected).toEqual(mockIndex);
    });
  });

  describe("Tabs Panel", () => {
    let TabsPanelWrapper;
    let mockName = "mock tab name";

    beforeEach(() => {
      TabsPanelWrapper = shallow(
        <TabsPanel name={mockName}>
          <p>Tab Panel</p>
        </TabsPanel>
      );
    });

    it("should match the snapshot", () => {
      expect(TabsPanelWrapper).toMatchSnapshot();
    });
  });
});
