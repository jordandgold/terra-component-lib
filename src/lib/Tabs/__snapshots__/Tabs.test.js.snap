// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Tabs Panel should match the snapshot 1`] = `
<div>
  <p>
    Tab Panel
  </p>
</div>
`;

exports[`Tabs should match the snapshot 1`] = `
<nav
  className="ter-tabs "
  data-controls="#tabs-one"
  role="navigation"
>
  <ul
    className="ter-tabs__list"
    role="tablist"
  >
    <li
      className="ter-tabs__list-item "
      key="0"
    >
      <button
        onClick={[Function]}
      >
        First
      </button>
    </li>
    <li
      className="ter-tabs__list-item is-selected"
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Second
      </button>
    </li>
    <li
      className="ter-tabs__list-item "
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Third
      </button>
    </li>
  </ul>
  <div
    className="ter-tabs__content"
  >
    <TabsPanel
      name="Second"
    >
      <h3>
        Second Tab
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit sollicitudin.
      </p>
    </TabsPanel>
  </div>
</nav>
`;
